e2e/onprem_test.go

```diff
@@ -1,13 +1,127 @@
 package e2e
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
+
 	"github.com/gravitational/robotest/e2e/framework"
-	"github.com/gravitational/robotest/e2e/specs"
+
+	"github.com/gravitational/robotest/e2e/uimodel"
+	uidefaults "github.com/gravitational/robotest/e2e/uimodel/defaults"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 )
 
 var _ = framework.RoboDescribe("Onprem Integration Test", func() {
 	f := framework.New()
-	specs.VerifyOnpremInstall(f)
-	specs.VerifyOnpremSite(f)
-	specs.VerifySiteUninstall(f)
+	ctx := framework.TestContext
+	var ui uimodel.UI
+	var domainName string
+
+	BeforeEach(func() {
+		domainName = ctx.ClusterName
+		ui = uimodel.Init(f.Page)
+	})
+
+	It("should provision a new cluster [provisioner:onprem_install]", func() {
```

Comment:
I thought we discussed splitting that into dedicated functions so that tests could be composed out of them